webapp/javascript/services/apps.ts

```diff
@@ -0,0 +1,37 @@
+import { Apps, appsModel } from '@webapp/models/app';
+import { Result } from '@webapp/util/fp';
+import type { ZodError } from 'zod';
+import type { RequestError } from './base';
+import { parseResponse, request } from './base';
+
+export interface FetchAppsError {
+  message?: string;
+}
+
+export async function fetchApps(): Promise<
+  Result<Apps, RequestError | ZodError>
+> {
+  const response = await request('/api/apps');
+
+  if (response.isOk) {
+    return parseResponse(response, appsModel);
+  }
+
+  return Result.err<Apps, RequestError>(response.error);
+}
+
+export async function deleteApp(data: {
+  name: string;
+}): Promise<Result<boolean, RequestError | ZodError>> {
+  const { name } = data;
+  const response = await request(`/api/apps`, {
+    method: 'DELETE',
+    body: JSON.stringify({ name }),
+  });
```

Comment:
This can take a long time (there are instances with 15+ minutes). How much is the timeout currently?

Reply:
We are using `fetch` for request, default timeouts is 300s in Chrome and 90s in Firefox and looks like there is no way to increase this values. Possible fix may be switching to axios or similar lib